`SIGBUS` trying to highlight a Standard ML file

### Summary

Zed crashes with a `SIGBUS` while trying to perform syntax highlighting on this code:

```sml
type a = { a : a,
           a : a,
           a : a,
           a : a,
           a : a,
           a : a };
```

(Minimised from https://github.com/CakeML/hardware/blob/6da9399c3b93ddcefc3a3bc46f5ffeea7faeb0e8/translator/translatorExpLib.sml.)

<details>
<summary>Backtrace (from be26accccabc58fc7b60195a3655e09ce064c0a7 for debug info):</summary>

```
* thread #1, name = 'main', queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x112b8d73f4)
  * frame #0: 0x0000000107219528 zed`ts_query_cursor_next_capture(self=0x0000600001f88be0, match=0x000000016fd9fa88, capture_index=0x000000016fd9fa84) at query.c:4132:61
    frame #1: 0x0000000106a5b854 zed`_$LT$tree_sitter..QueryCaptures$LT$T$C$I$GT$$u20$as$u20$core..iter..traits..iterator..Iterator$GT$::next::h9a1f86cda1d1a510 at lib.rs:2683:20
    frame #2: 0x0000000106911d8c zed`language::syntax_map::SyntaxMapCapturesLayer::advance::hfd32e9bd070e7eb5 at syntax_map.rs:1146:29
    frame #3: 0x0000000106a5f760 zed`language::syntax_map::SyntaxMapCaptures::new::h8e052ade3a902a87 at syntax_map.rs:947:13
    frame #4: 0x0000000106911478 zed`language::syntax_map::SyntaxSnapshot::captures::hfd3802000e7fa8d0 at syntax_map.rs:798:9
    frame #5: 0x0000000106a20bc0 zed`language::buffer::BufferSnapshot::get_highlights::h22c7da88e51ae627 at buffer.rs:3043:24
    frame #6: 0x00000001069ed68c zed`language::buffer::BufferSnapshot::chunks::h505fcf4e79ee98ba at buffer.rs:3063:27
    frame #7: 0x000000010442b654 zed`multi_buffer::Excerpt::chunks_in_range::hfa335c3ceb7abc55 at multi_buffer.rs:6463:30
    frame #8: 0x000000010442cdf8 zed`multi_buffer::MultiBufferChunks::seek_to_excerpt_offset_range::h80faf5cd8802b1c4 at multi_buffer.rs:7077:44
    frame #9: 0x000000010442cbc0 zed`multi_buffer::MultiBufferChunks::seek::hb48b83b7d26e6cc4 at multi_buffer.rs:7056:9
    frame #10: 0x000000010443e778 zed`multi_buffer::MultiBufferSnapshot::chunks::hac7532f74dcc9118 at multi_buffer.rs:3993:9
    frame #11: 0x0000000103aa3078 zed`editor::display_map::custom_highlights::CustomHighlightsChunks::new::h77f0efa75ff70f92 at custom_highlights.rs:42:28
    frame #12: 0x0000000103aadd1c zed`editor::display_map::inlay_map::InlaySnapshot::chunks::he3b1ddcd2bce0ab6 at inlay_map.rs:997:29
    frame #13: 0x0000000103aa6d94 zed`editor::display_map::fold_map::FoldSnapshot::chunks::h28479ab86cecd8b0 at fold_map.rs:794:27
    frame #14: 0x0000000103aaf9a0 zed`editor::display_map::tab_map::TabSnapshot::chunks::h18cd481a65239619 at tab_map.rs:255:26
    frame #15: 0x0000000103ab3524 zed`editor::display_map::wrap_map::WrapSnapshot::chunks::hee4f1c452f78c45d at wrap_map.rs:604:27
    frame #16: 0x000000010428a54c zed`editor::display_map::block_map::BlockSnapshot::chunks::h24686c49e20704cc at block_map.rs:1350:27
    frame #17: 0x0000000103aba608 zed`editor::display_map::DisplaySnapshot::chunks::hff885e34e7fdf631 at display_map.rs:870:9
    frame #18: 0x0000000103aba70c zed`editor::display_map::DisplaySnapshot::highlighted_chunks::h4de611a6e193d81a at display_map.rs:888:9
    frame #19: 0x0000000103af04cc zed`editor::element::layout_line::h41b0ddf54147fcc1 at element.rs:7963:18
    frame #20: 0x0000000103ae9a7c zed`_$LT$editor..element..EditorElement$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hf2b95d02784abac0 at element.rs:7023:46
    frame #21: 0x0000000103d5f140 zed`gpui::window::Window::with_content_mask::hf321216fb6d9c7aa at window.rs:1951:26
    frame #22: 0x0000000103ae8044 zed`_$LT$editor..element..EditorElement$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h50b11af4b7f400c5 at element.rs:6709:17
    frame #23: 0x0000000103d5eae0 zed`gpui::window::Window::with_text_style::hcef9ec0a796c48e6 at window.rs:1911:26
    frame #24: 0x0000000103ae7fd8 zed`_$LT$editor..element..EditorElement$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h9ca3fa8fb4a122d4 at element.rs:6708:13
    frame #25: 0x0000000103d5d3bc zed`gpui::window::Window::with_rem_size::h47d57c9a6c5366d3 at window.rs:1460:26
    frame #26: 0x0000000103ae7f1c zed`_$LT$editor..element..EditorElement$u20$as$u20$gpui..element..Element$GT$::prepaint::hdc10539b2a6224d1 at element.rs:6707:9
    frame #27: 0x0000000103cbb890 zed`gpui::element::Drawable$LT$E$GT$::prepaint::h38ac297f85697812 at element.rs:338:32
    frame #28: 0x0000000103cef5f0 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::h466f7fdf1bdf0efe at element.rs:466:9
    frame #29: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #30: 0x0000000107ecc700 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h3c4125305ff95e0a at view.rs:234:17
    frame #31: 0x0000000107df04a8 zed`gpui::window::Window::with_rendered_view::h84ce848c3a67716f at window.rs:2824:22
    frame #32: 0x0000000107ecbf54 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::h34136c8e9724bb97 at view.rs:233:13
    frame #33: 0x0000000107eac2a0 zed`gpui::element::Drawable$LT$E$GT$::prepaint::hca7f534e18dd025c at element.rs:338:32
    frame #34: 0x0000000107eafe10 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::h3aea0ab97dbef344 at element.rs:466:9
    frame #35: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #36: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #37: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #38: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #39: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #40: 0x0000000107ddb910 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1951:26
    frame #41: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #42: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #43: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #44: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #45: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #46: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #47: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #48: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #49: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #50: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #51: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #52: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #53: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #54: 0x0000000107ddb910 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1951:26
    frame #55: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #56: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #57: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #58: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #59: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #60: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #61: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #62: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #63: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #64: 0x0000000107eb0f90 zed`gpui::element::AnyElement::prepaint_at::_$u7b$$u7b$closure$u7d$$u7d$::h9b467866564f054b at element.rs:548:62
    frame #65: 0x0000000107ddcf08 zed`gpui::window::Window::with_absolute_element_offset::h1ebb2f4e9b71f626 at window.rs:1986:22
    frame #66: 0x00000001080526dc zed`gpui::element::AnyElement::prepaint_at::h4206cfab09eeea19 at element.rs:548:9
    frame #67: 0x0000000107ecc694 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hfe9d524437c6d138 at view.rs:208:33
    frame #68: 0x0000000107da6530 zed`gpui::app::App::detect_accessed_entities::h2092019fe746582f at app.rs:449:22
    frame #69: 0x0000000107ecc5d0 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hca11085c7b617af7 at view.rs:205:29
    frame #70: 0x0000000107df03e8 zed`gpui::window::Window::with_rendered_view::h05aeabed3fa36389 at window.rs:2824:22
    frame #71: 0x0000000107ecc300 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h9455d4beedb7bb42 at view.rs:204:25
    frame #72: 0x0000000107ddf260 zed`gpui::window::Window::with_element_state::h12b7d18a5f84e5db at window.rs:2178:35
    frame #73: 0x0000000107ecbf1c zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::h34136c8e9724bb97 at view.rs:177:13
    frame #74: 0x0000000107eac2a0 zed`gpui::element::Drawable$LT$E$GT$::prepaint::hca7f534e18dd025c at element.rs:338:32
    frame #75: 0x0000000107eafe10 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::h3aea0ab97dbef344 at element.rs:466:9
    frame #76: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #77: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #78: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #79: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #80: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #81: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #82: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #83: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #84: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #85: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #86: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #87: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #88: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #89: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #90: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #91: 0x0000000108052790 zed`_$LT$gpui..element..AnyElement$u20$as$u20$gpui..element..Element$GT$::prepaint::h52b8166fea65c068 at element.rs:591:9
    frame #92: 0x0000000104937f88 zed`gpui::element::Drawable$LT$E$GT$::prepaint::h429d9dbdf6eb3dda at element.rs:338:32
    frame #93: 0x0000000104944a5c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hbbf7e3982769d013 at element.rs:466:9
    frame #94: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #95: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #96: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #97: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #98: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #99: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #100: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #101: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #102: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #103: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #104: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #105: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #106: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #107: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #108: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #109: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #110: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #111: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #112: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #113: 0x0000000107ddb910 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1951:26
    frame #114: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #115: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #116: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #117: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #118: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #119: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #120: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #121: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #122: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #123: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #124: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #125: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #126: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #127: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #128: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #129: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #130: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #131: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #132: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #133: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #134: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #135: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #136: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #137: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #138: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #139: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #140: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #141: 0x0000000107ddb910 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1951:26
    frame #142: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #143: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #144: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #145: 0x0000000107dec684 zed`gpui::window::Window::with_optional_element_state::_$u7b$$u7b$closure$u7d$$u7d$::h3f5cb95c3452d07e at window.rs:2221:39
    frame #146: 0x0000000107de0ed4 zed`gpui::window::Window::with_element_state::h5d2d4094eb8a6753 at window.rs:2178:35
    frame #147: 0x0000000107deb534 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2220:13
    frame #148: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #149: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #150: 0x0000000107ebf6cc zed`_$LT$gpui..elements..div..Stateful$LT$E$GT$$u20$as$u20$gpui..element..Element$GT$::prepaint::hb094062e54502cec at div.rs:2785:9
    frame #151: 0x0000000107eab0f4 zed`gpui::element::Drawable$LT$E$GT$::prepaint::h23893c4e176b3245 at element.rs:338:32
    frame #152: 0x0000000107eafe38 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::ha3192840a12ab13e at element.rs:466:9
    frame #153: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #154: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #155: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #156: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #157: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #158: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #159: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #160: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #161: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #162: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #163: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #164: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #165: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #166: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #167: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #168: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #169: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #170: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #171: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #172: 0x0000000107ddb910 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1951:26
    frame #173: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #174: 0x0000000107dd93b4 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1911:26
    frame #175: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #176: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #177: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #178: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #179: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #180: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #181: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #182: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #183: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #184: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #185: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #186: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #187: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #188: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #189: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #190: 0x0000000107deb590 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2227:35
    frame #191: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #192: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #193: 0x0000000107eab71c zed`gpui::element::Drawable$LT$E$GT$::prepaint::h36470d0b3d08942b at element.rs:338:32
    frame #194: 0x0000000107eafe4c zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::hfa38288939283744 at element.rs:466:9
    frame #195: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #196: 0x0000000107eb2780 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h34936f3313dde8f6 at div.rs:1281:25
    frame #197: 0x0000000107ddce14 zed`gpui::window::Window::with_element_offset::h9e0baeca18719cc8 at window.rs:1969:20
    frame #198: 0x0000000107eb2618 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h6241703a0887b7a9 at div.rs:1279:17
    frame #199: 0x0000000107eb5f00 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h3fd98b91469d0a69 at div.rs:1495:42
    frame #200: 0x0000000107ddb8b8 zed`gpui::window::Window::with_content_mask::h7b6ecd393e154251 at window.rs:1955:13
    frame #201: 0x0000000107eb58b4 zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h0591d787e2d61a87 at div.rs:1484:21
    frame #202: 0x0000000107dd9384 zed`gpui::window::Window::with_text_style::h4bd02f494c715dcd at window.rs:1915:13
    frame #203: 0x0000000107eb525c zed`gpui::elements::div::Interactivity::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h75507c41c6f33572 at div.rs:1483:17
    frame #204: 0x0000000107dec684 zed`gpui::window::Window::with_optional_element_state::_$u7b$$u7b$closure$u7d$$u7d$::h3f5cb95c3452d07e at window.rs:2221:39
    frame #205: 0x0000000107de0ed4 zed`gpui::window::Window::with_element_state::h5d2d4094eb8a6753 at window.rs:2178:35
    frame #206: 0x0000000107deb534 zed`gpui::window::Window::with_optional_element_state::hac7caefc5da05f1b at window.rs:2220:13
    frame #207: 0x0000000107eb474c zed`gpui::elements::div::Interactivity::prepaint::h77df71bf4d81f82f at div.rs:1461:9
    frame #208: 0x0000000107eb2230 zed`_$LT$gpui..elements..div..Div$u20$as$u20$gpui..element..Element$GT$::prepaint::h10bd4fce705ca408 at div.rs:1272:9
    frame #209: 0x0000000107ebf6cc zed`_$LT$gpui..elements..div..Stateful$LT$E$GT$$u20$as$u20$gpui..element..Element$GT$::prepaint::hb094062e54502cec at div.rs:2785:9
    frame #210: 0x0000000107eab0f4 zed`gpui::element::Drawable$LT$E$GT$::prepaint::h23893c4e176b3245 at element.rs:338:32
    frame #211: 0x0000000107eafe38 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::ha3192840a12ab13e at element.rs:466:9
    frame #212: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #213: 0x0000000107ecc700 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::_$u7b$$u7b$closure$u7d$$u7d$::h3c4125305ff95e0a at view.rs:234:17
    frame #214: 0x0000000107df04a8 zed`gpui::window::Window::with_rendered_view::h84ce848c3a67716f at window.rs:2824:22
    frame #215: 0x0000000107ecbf54 zed`_$LT$gpui..view..AnyView$u20$as$u20$gpui..element..Element$GT$::prepaint::h34136c8e9724bb97 at view.rs:233:13
    frame #216: 0x0000000107eac2a0 zed`gpui::element::Drawable$LT$E$GT$::prepaint::hca7f534e18dd025c at element.rs:338:32
    frame #217: 0x0000000107eafe10 zed`_$LT$gpui..element..Drawable$LT$E$GT$$u20$as$u20$gpui..element..ElementObject$GT$::prepaint::h3aea0ab97dbef344 at element.rs:466:9
    frame #218: 0x00000001080525c4 zed`gpui::element::AnyElement::prepaint::hea4d4c79cedff040 at element.rs:514:9
    frame #219: 0x0000000107eb0fc4 zed`gpui::element::AnyElement::prepaint_as_root::_$u7b$$u7b$closure$u7d$$u7d$::h66aaa6e2bf95bdd7 at element.rs:561:62
    frame #220: 0x0000000107ddd0e0 zed`gpui::window::Window::with_absolute_element_offset::h62402978a05152da at window.rs:1986:22
    frame #221: 0x0000000108052738 zed`gpui::element::AnyElement::prepaint_as_root::hdd74da8cb7e36f97 at element.rs:561:9
    frame #222: 0x0000000107dd6e04 zed`gpui::window::Window::draw_roots::hedc8411fe3ef8ac3 at window.rs:1626:9
    frame #223: 0x0000000107dfa418 zed`gpui::window::Window::draw::h3821d263aacbb1fd at window.rs:1527:9
    frame #224: 0x0000000107dd4438 zed`gpui::window::Window::new::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hec865e69dc567811 at window.rs:801:33
    frame #225: 0x0000000107db1158 zed`gpui::app::App::update_window_id::_$u7b$$u7b$closure$u7d$$u7d$::hc5ce84b50911b610 at app.rs:1005:26
    frame #226: 0x0000000107da5314 zed`gpui::app::App::update::hb7d67b38a990b103 at app.rs:419:22
    frame #227: 0x0000000107da8c4c zed`gpui::app::App::update_window_id::h87a652d63d025c26 at app.rs:994:9
    frame #228: 0x0000000107db8828 zed`_$LT$gpui..app..App$u20$as$u20$gpui..AppContext$GT$::update_window::h6e9cbcc17381e946 at app.rs:1653:9
    frame #229: 0x0000000107d9fdfc zed`_$LT$gpui..app..async_context..AsyncApp$u20$as$u20$gpui..AppContext$GT$::update_window::hf9c002543df16330 at async_context.rs:92:9
    frame #230: 0x0000000107df7f7c zed`gpui::window::AnyWindowHandle::update::h0695046434494d84 at window.rs:3928:9
    frame #231: 0x0000000107dd4400 zed`gpui::window::Window::new::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::hbdc915de3396a155 at window.rs:799:25
    frame #232: 0x000000010800a764 zed`util::measure::hc8d24e9d070ad58d at util.rs:386:9
    frame #233: 0x0000000107dd41f4 zed`gpui::window::Window::new::_$u7b$$u7b$closure$u7d$$u7d$::hb07fbe6133397c7b at window.rs:798:21
    frame #234: 0x0000000107f5d988 zed`_$LT$alloc..boxed..Box$LT$F$C$A$GT$$u20$as$u20$core..ops..function..FnMut$LT$Args$GT$$GT$::call_mut::hdb7b2b414f275f74 at boxed.rs:2077:9
    frame #235: 0x0000000108163ba8 zed`gpui::platform::mac::window::step::h1b7c1c95d228c4b1 at window.rs:1657:9
    frame #236: 0x000000018c4b15b4 libdispatch.dylib`_dispatch_client_callout + 20
    frame #237: 0x000000018c4b4a68 libdispatch.dylib`_dispatch_continuation_pop + 596
    frame #238: 0x000000018c4c8e84 libdispatch.dylib`_dispatch_source_latch_and_call + 420
    frame #239: 0x000000018c4c7a4c libdispatch.dylib`_dispatch_source_invoke + 836
    frame #240: 0x000000018c4bff54 libdispatch.dylib`_dispatch_main_queue_drain + 748
    frame #241: 0x000000018c4bfc58 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 44
    frame #242: 0x000000018c78b9c0 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
    frame #243: 0x000000018c74b5ac CoreFoundation`__CFRunLoopRun + 1996
    frame #244: 0x000000018c74a724 CoreFoundation`CFRunLoopRunSpecific + 588
    frame #245: 0x0000000197ca2530 HIToolbox`RunCurrentEventLoopInMode + 292
    frame #246: 0x0000000197ca8348 HIToolbox`ReceiveNextEventCommon + 676
    frame #247: 0x0000000197ca8508 HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 76
    frame #248: 0x00000001902b5034 AppKit`_DPSNextEvent + 660
    frame #249: 0x0000000190c192d4 AppKit`-[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 688
    frame #250: 0x00000001902a8060 AppKit`-[NSApplication run] + 480
    frame #251: 0x0000000108361a90 zed`_$LT$$LP$$RP$$u20$as$u20$objc..message..MessageArguments$GT$::invoke::ha864c21503a9e9d1 at mod.rs:128:17
    frame #252: 0x0000000108360c08 zed`objc::message::platform::send_unverified::hc97e701b70acddb8 at mod.rs:27:9
    frame #253: 0x00000001083554c4 zed`_$LT$$BP$mut$u20$objc..runtime..Object$u20$as$u20$cocoa..appkit..NSApplication$GT$::run::h360dd79d98ab63ef [inlined] objc::message::send_message::hfc4c6a46186904c3 at mod.rs:178:5
    frame #254: 0x00000001083554a8 zed`_$LT$$BP$mut$u20$objc..runtime..Object$u20$as$u20$cocoa..appkit..NSApplication$GT$::run::h360dd79d98ab63ef at appkit.rs:628:9
    frame #255: 0x000000010814d528 zed`_$LT$gpui..platform..mac..platform..MacPlatform$u20$as$u20$gpui..platform..Platform$GT$::run::hce68852b608ad052 at platform.rs:458:13
    frame #256: 0x00000001001a8874 zed`gpui::app::Application::run::h3d475ad46afcfcdc at app.rs:164:9
    frame #257: 0x000000010026afb8 zed`zed::main::h311ee39a08e1f191 at main.rs:284:5
    frame #258: 0x0000000100168890 zed`core::ops::function::FnOnce::call_once::h5161602b7911a7a6 at function.rs:250:5
    frame #259: 0x000000010006b8b4 zed`std::sys::backtrace::__rust_begin_short_backtrace::h47c75f3e3b2540b0 at backtrace.rs:152:18
    frame #260: 0x000000010002f05c zed`std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::h2651fdde21a58ca8 at rt.rs:162:18
    frame #261: 0x0000000108be4468 zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] core::ops::function::impls::_$LT$impl$u20$core..ops..function..FnOnce$LT$A$GT$$u20$for$u20$$RF$F$GT$::call_once::h4f74490c6170ea16 at function.rs:284:13 [opt]
    frame #262: 0x0000000108be4460 zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panicking::try::do_call::h2f36d2f1f1af8d28 at panicking.rs:557:40 [opt]
    frame #263: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panicking::try::ha6af9029a7d93c94 at panicking.rs:521:19 [opt]
    frame #264: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panic::catch_unwind::ha4f738ae2ba7c3a4 at panic.rs:350:14 [opt]
    frame #265: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::rt::lang_start_internal::_$u7b$$u7b$closure$u7d$$u7d$::hf216622dc2c733e3 at rt.rs:141:48 [opt]
    frame #266: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panicking::try::do_call::haa691957db1dd55f at panicking.rs:557:40 [opt]
    frame #267: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panicking::try::ha0c1a49b9fabc98e at panicking.rs:521:19 [opt]
    frame #268: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 [inlined] std::panic::catch_unwind::h68ad032c646bb0ab at panic.rs:350:14 [opt]
    frame #269: 0x0000000108be445c zed`std::rt::lang_start_internal::hdd117cb81a316264 at rt.rs:141:20 [opt]
    frame #270: 0x000000010002f034 zed`std::rt::lang_start::h34d820e1e1d82056 at rt.rs:161:17
    frame #271: 0x0000000100270634 zed`main + 36
    frame #272: 0x000000018c2e4274 dyld`start + 2840
```

</details>

I think this issue was fixed in https://github.com/tree-sitter/tree-sitter/pull/3727, so a tree-sitter upgrade should resolve it.

<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Save the above code to a `.sml` file, say `repro.sml`.
2. Install the 'Standard ML' extension.
3. Open that file.

Actual Behavior:
Zed crashes. If I run Zed's main binary instead of the CLI, I can see that it's a `SIGBUS` error.

Expected Behavior:
The file is opened successfully.

### Zed Version and System Specs

Zed: v0.172.10 (Zed)
OS: macOS 15.2.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2025-02-08T11:54:09.477176+11:00 [INFO] ========== starting zed ==========
2025-02-08T11:54:09.542649+11:00 [INFO] Opening main db
2025-02-08T11:54:09.54382+11:00 [INFO] Opening main db
2025-02-08T11:54:09.557922+11:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2025-02-08T11:54:09.68867+11:00 [INFO] set environment variables from shell:/run/current-system/sw/bin/fish, path:/Users/liam/Library/pnpm:/Users/liam/src/PipelineC/src:/Users/liam/.deno/bin:/Users/liam/.local/bin:/Users/liam/.cargo/bin:/Users/liam/.nix-profile/bin:/run/current-system/sw/bin:/nix/var/nix/profiles/default/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
2025-02-08T11:54:09.766426+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.767097+11:00 [INFO] No prompt template overrides directory found at /Users/liam/.config/zed/prompt_overrides. Using built-in prompts.
2025-02-08T11:54:09.767701+11:00 [INFO] extensions updated. loading 18, reloading 0, unloading 0
2025-02-08T11:54:09.806465+11:00 [INFO] Opening main db
2025-02-08T11:54:09.811651+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.81372+11:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2025-02-08T11:54:09.813828+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.83844+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.839988+11:00 [INFO] Opening main db
2025-02-08T11:54:09.847172+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.87182+11:00 [INFO] set status on client 0: Authenticating
2025-02-08T11:54:09.877676+11:00 [INFO] set status on client 46063: Connecting
2025-02-08T11:54:09.914215+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:09.917966+11:00 [ERROR] theme not found: Xcode Default Darker
2025-02-08T11:54:10.150213+11:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-02-08T11:54:10.999757+11:00 [INFO] add_connection;
2025-02-08T11:54:11.010899+11:00 [INFO] set status on client 46063: Connected { peer_id: PeerId { owner_id: 677, id: 1828133 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2025-02-08T11:54:11.378612+11:00 [INFO] using project environment variables shell launched in "/Users/liam/src/zed". PATH="/Users/liam/Library/pnpm:/Users/liam/src/PipelineC/src:/Users/liam/.deno/bin:/Users/liam/.local/bin:/Users/liam/.cargo/bin:/Users/liam/.nix-profile/bin:/run/current-system/sw/bin:/nix/var/nix/profiles/default/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
2025-02-08T11:54:12.237416+11:00 [INFO] Opening main db
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -499,6 +499,7 @@ simplelog = "0.12.2"
 smallvec = { version = "1.6", features = ["union"] }
 smol = "2.0"
 sqlformat = "0.2"
+streaming-iterator = "0.1"
 strsim = "0.11"
 strum = { version = "0.26.0", features = ["derive"] }
 subtle = "2.5.0"
@@ -520,7 +521,7 @@ tiny_http = "0.8"
 toml = "0.8"
 tokio = { version = "1" }
 tower-http = "0.4.4"
-tree-sitter = { version = "0.23", features = ["wasm"] }
+tree-sitter = { version = "0.24", features = ["wasm"] }
 tree-sitter-bash = "0.23"
 tree-sitter-c = "0.23"
 tree-sitter-cpp = "0.23"
@@ -549,16 +550,16 @@ unicode-segmentation = "1.10"
 unicode-script = "0.5.7"
 url = "2.2"
 uuid = { version = "1.1.2", features = ["v4", "v5", "v7", "serde"] }
-wasmparser = "0.215"
-wasm-encoder = "0.215"
-wasmtime = { version = "24", default-features = false, features = [
+wasmparser = "0.217"
+wasm-encoder = "0.217"
+wasmtime = { version = "25", default-features = false, features = [
     "async",
     "demangle",
     "runtime",
     "cranelift",
     "component-model",
 ] }
-wasmtime-wasi = "24"
+wasmtime-wasi = "25"
 which = "6.0.0"
 wit-component = "0.201"
 zed_llm_client = "0.4"

--- a/crates/editor/src/editor_tests.rs
+++ b/crates/editor/src/editor_tests.rs
@@ -9711,7 +9711,7 @@ async fn test_toggle_block_comment(cx: &mut gpui::TestAppContext) {
         &r#"
             <!-- ˇ<script> -->
                 // ˇvar x = new Y();
-            // ˇ</script>
+            <!-- ˇ</script> -->
         "#
         .unindent(),
     );

--- a/crates/extension_host/src/wasm_host/wit/since_v0_0_1.rs
+++ b/crates/extension_host/src/wasm_host/wit/since_v0_0_1.rs
@@ -84,7 +84,7 @@ impl HostWorktree for WasmState {
         latest::HostWorktree::which(self, delegate, binary_name).await
     }
 
-    fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
         Ok(())
     }
 }

--- a/crates/extension_host/src/wasm_host/wit/since_v0_0_4.rs
+++ b/crates/extension_host/src/wasm_host/wit/since_v0_0_4.rs
@@ -92,7 +92,7 @@ impl HostWorktree for WasmState {
         latest::HostWorktree::which(self, delegate, binary_name).await
     }
 
-    fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
         // We only ever hand out borrows of worktrees.
         Ok(())
     }

--- a/crates/extension_host/src/wasm_host/wit/since_v0_0_6.rs
+++ b/crates/extension_host/src/wasm_host/wit/since_v0_0_6.rs
@@ -147,7 +147,7 @@ impl HostWorktree for WasmState {
         latest::HostWorktree::which(self, delegate, binary_name).await
     }
 
-    fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
         // We only ever hand out borrows of worktrees.
         Ok(())
     }

--- a/crates/extension_host/src/wasm_host/wit/since_v0_1_0.rs
+++ b/crates/extension_host/src/wasm_host/wit/since_v0_1_0.rs
@@ -240,7 +240,7 @@ impl HostKeyValueStore for WasmState {
         kv_store.insert(key, value).await.to_wasmtime_result()
     }
 
-    fn drop(&mut self, _worktree: Resource<ExtensionKeyValueStore>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<ExtensionKeyValueStore>) -> Result<()> {
         // We only ever hand out borrows of key-value stores.
         Ok(())
     }
@@ -282,7 +282,7 @@ impl HostWorktree for WasmState {
         latest::HostWorktree::which(self, delegate, binary_name).await
     }
 
-    fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
         // We only ever hand out borrows of worktrees.
         Ok(())
     }
@@ -350,7 +350,7 @@ impl http_client::HostHttpResponseStream for WasmState {
         .to_wasmtime_result()
     }
 
-    fn drop(&mut self, _resource: Resource<ExtensionHttpResponseStream>) -> Result<()> {
+    async fn drop(&mut self, _resource: Resource<ExtensionHttpResponseStream>) -> Result<()> {
         Ok(())
     }
 }

--- a/crates/extension_host/src/wasm_host/wit/since_v0_2_0.rs
+++ b/crates/extension_host/src/wasm_host/wit/since_v0_2_0.rs
@@ -259,7 +259,7 @@ impl HostKeyValueStore for WasmState {
         kv_store.insert(key, value).await.to_wasmtime_result()
     }
 
-    fn drop(&mut self, _worktree: Resource<ExtensionKeyValueStore>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<ExtensionKeyValueStore>) -> Result<()> {
         // We only ever hand out borrows of key-value stores.
         Ok(())
     }
@@ -275,7 +275,7 @@ impl HostProject for WasmState {
         Ok(project.worktree_ids())
     }
 
-    fn drop(&mut self, _project: Resource<Project>) -> Result<()> {
+    async fn drop(&mut self, _project: Resource<Project>) -> Result<()> {
         // We only ever hand out borrows of projects.
         Ok(())
     }
@@ -325,7 +325,7 @@ impl HostWorktree for WasmState {
         Ok(delegate.which(binary_name).await)
     }
 
-    fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
+    async fn drop(&mut self, _worktree: Resource<Worktree>) -> Result<()> {
         // We only ever hand out borrows of worktrees.
         Ok(())
     }
@@ -393,7 +393,7 @@ impl http_client::HostHttpResponseStream for WasmState {
         .to_wasmtime_result()
     }
 
-    fn drop(&mut self, _resource: Resource<ExtensionHttpResponseStream>) -> Result<()> {
+    async fn drop(&mut self, _resource: Resource<ExtensionHttpResponseStream>) -> Result<()> {
         Ok(())
     }
 }

--- a/crates/language/Cargo.toml
+++ b/crates/language/Cargo.toml
@@ -53,6 +53,7 @@ settings.workspace = true
 similar.workspace = true
 smallvec.workspace = true
 smol.workspace = true
+streaming-iterator.workspace = true
 strsim.workspace = true
 sum_tree.workspace = true
 task.workspace = true

--- a/crates/language/src/syntax_map.rs
+++ b/crates/language/src/syntax_map.rs
@@ -14,6 +14,7 @@ use std::{
     ops::{Deref, DerefMut, Range},
     sync::Arc,
 };
+use streaming_iterator::StreamingIterator;
 use sum_tree::{Bias, SeekTarget, SumTree};
 use text::{Anchor, BufferSnapshot, OffsetRangeExt, Point, Rope, ToOffset, ToPoint};
 use tree_sitter::{Node, Query, QueryCapture, QueryCaptures, QueryCursor, QueryMatches, Tree};
@@ -1143,7 +1144,7 @@ impl<'a> SyntaxMapMatches<'a> {
 
 impl<'a> SyntaxMapCapturesLayer<'a> {
     fn advance(&mut self) {
-        self.next_capture = self.captures.next().map(|(mat, ix)| mat.captures[ix]);
+        self.next_capture = self.captures.next().map(|(mat, ix)| mat.captures[*ix]);
     }
 
     fn sort_key(&self) -> (usize, Reverse<usize>, usize) {
@@ -1280,7 +1281,8 @@ fn get_injections(
 
     for query_range in changed_ranges {
         query_cursor.set_byte_range(query_range.start.saturating_sub(1)..query_range.end + 1);
-        for mat in query_cursor.matches(&config.query, node, TextProvider(text.as_rope())) {
+        let mut matches = query_cursor.matches(&config.query, node, TextProvider(text.as_rope()));
+        while let Some(mat) = matches.next() {
             let content_ranges = mat
                 .nodes_for_capture_index(config.content_capture_ix)
                 .map(|node| node.range())
@@ -1554,7 +1556,8 @@ impl<'a> SyntaxLayer<'a> {
         query_cursor.set_byte_range(offset.saturating_sub(1)..offset.saturating_add(1));
 
         let mut smallest_match: Option<(u32, Range<usize>)> = None;
-        for mat in query_cursor.matches(&config.query, self.node(), text) {
+        let mut matches = query_cursor.matches(&config.query, self.node(), text);
+        while let Some(mat) = matches.next() {
             for capture in mat.captures {
                 let Some(override_entry) = config.values.get(&capture.index) else {
                     continue;

--- a/crates/migrator/Cargo.toml
+++ b/crates/migrator/Cargo.toml
@@ -15,6 +15,7 @@ doctest = false
 [dependencies]
 collections.workspace = true
 convert_case.workspace = true
+streaming-iterator.workspace = true
 tree-sitter-json.workspace = true
 tree-sitter.workspace = true
 

--- a/crates/migrator/src/migrator.rs
+++ b/crates/migrator/src/migrator.rs
@@ -1,6 +1,7 @@
 use collections::HashMap;
 use convert_case::{Case, Casing};
 use std::{cmp::Reverse, ops::Range, sync::LazyLock};
+use streaming_iterator::StreamingIterator;
 use tree_sitter::{Query, QueryMatch};
 
 fn migrate(text: &str, patterns: MigrationPatterns, query: &Query) -> Option<String> {
@@ -11,10 +12,10 @@ fn migrate(text: &str, patterns: MigrationPatterns, query: &Query) -> Option<Str
     let syntax_tree = parser.parse(&text, None).unwrap();
 
     let mut cursor = tree_sitter::QueryCursor::new();
-    let matches = cursor.matches(query, syntax_tree.root_node(), text.as_bytes());
+    let mut matches = cursor.matches(query, syntax_tree.root_node(), text.as_bytes());
 
     let mut edits = vec![];
-    for mat in matches {
+    while let Some(mat) = matches.next() {
         if let Some((_, callback)) = patterns.get(mat.pattern_index) {
             edits.extend(callback(&text, &mat, query));
         }

--- a/crates/semantic_index/Cargo.toml
+++ b/crates/semantic_index/Cargo.toml
@@ -42,6 +42,7 @@ serde_json.workspace = true
 settings.workspace = true
 sha2.workspace = true
 smol.workspace = true
+streaming-iterator.workspace = true
 theme.workspace = true
 tree-sitter.workspace = true
 ui.workspace = true

--- a/crates/semantic_index/src/chunking.rs
+++ b/crates/semantic_index/src/chunking.rs
@@ -7,6 +7,7 @@ use std::{
     path::Path,
     sync::Arc,
 };
+use streaming_iterator::StreamingIterator;
 use tree_sitter::QueryCapture;
 use util::ResultExt as _;
 
@@ -88,7 +89,7 @@ fn syntactic_ranges(
     let mut ranges = with_query_cursor(|cursor| {
         cursor
             .matches(&outline.query, tree.root_node(), text.as_bytes())
-            .filter_map(|mat| {
+            .filter_map_deref(|mat| {
                 mat.captures
                     .iter()
                     .find_map(|QueryCapture { node, index }| {

--- a/crates/settings/Cargo.toml
+++ b/crates/settings/Cargo.toml
@@ -32,6 +32,7 @@ serde_derive.workspace = true
 serde_json.workspace = true
 serde_json_lenient.workspace = true
 smallvec.workspace = true
+streaming-iterator.workspace = true
 tree-sitter-json.workspace = true
 tree-sitter.workspace = true
 util.workspace = true

--- a/crates/settings/src/settings_store.rs
+++ b/crates/settings/src/settings_store.rs
@@ -17,6 +17,7 @@ use std::{
     str::{self, FromStr},
     sync::{Arc, LazyLock},
 };
+use streaming_iterator::StreamingIterator;
 use tree_sitter::Query;
 use util::RangeExt;
 
@@ -1263,8 +1264,8 @@ fn replace_value_in_json_text(
     let mut last_value_range = 0..0;
     let mut first_key_start = None;
     let mut existing_value_range = 0..text.len();
-    let matches = cursor.matches(&PAIR_QUERY, syntax_tree.root_node(), text.as_bytes());
-    for mat in matches {
+    let mut matches = cursor.matches(&PAIR_QUERY, syntax_tree.root_node(), text.as_bytes());
+    while let Some(mat) = matches.next() {
         if mat.captures.len() != 2 {
             continue;
         }

